lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -428,6 +428,19 @@ class KymoTrackGroup:
 
     def __init__(self, kymo_tracks):
         self._src = kymo_tracks
+        if self._src:
+            self._validate_compatibility(kymo_tracks, compare_to_self=False)
+
+    def _validate_compatibility(self, kymo_tracks, compare_to_self=False):
+        new_kymos = set([track._kymo._id for track in kymo_tracks])
+        new_channels = set([track._channel for track in kymo_tracks])
+
+        assert len(new_kymos) == 1 and (
+            self._kymo._id == list(new_kymos)[0] if compare_to_self else True
```

Comment:
You don't have to convert the set to a `list` to extract one element. Just do `next(iter(new_kymos))`. Same for channels.

Personally, I find the `compare_to_self` argument a bit smelly. You can just return the unique kymo and unique channel and throw if it's not unique. Then compare upstream to the kymo already associated with the group if need be. There's only one place where you actually need to do that.

Reply:
good points, I like it better now